ui/components/app/multichain-transaction-details-modal/multichain-transaction-details-modal.test.tsx

```diff
@@ -0,0 +1,157 @@
+      ticker: 'BTC',
+      nickname: 'Bitcoin',
+      isAddressCompatible: (_address: string) => true,
+      rpcPrefs: {
+        blockExplorerUrl: 'https://explorer.bitcoin.com',
+      },
+    },
+  },
+};
+
+describe('MultichainTransactionDetailsModal', () => {
+  const mockTrackEvent = jest.fn();
+  const useI18nContextMock = useI18nContext as jest.Mock;
+
+  beforeEach(() => {
+    useI18nContextMock.mockReturnValue((key: string) => key);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const renderComponent = (props = mockProps) => {
+    return renderWithProvider(
+      <MetaMetricsContext.Provider value={mockTrackEvent}>
+        <MultichainTransactionDetailsModal {...props} />
+      </MetaMetricsContext.Provider>,
+    );
+  };
+
+  it('renders the modal with transaction details', () => {
+    renderComponent();
+
+    expect(screen.getByText('Send btc')).toBeInTheDocument();
+    expect(screen.getByText('Confirmed')).toBeInTheDocument();
+    expect(screen.getByTestId('transaction-amount')).toHaveTextContent(
+      '1.2 BTC',
+    );
+  });
+
+  it('displays the correct transaction status with appropriate color', () => {
+    renderComponent();
+    const statusElement = screen.getByText('Confirmed');
+    expect(statusElement).toHaveClass('mm-box--color-success-default');
+  });
+
+  it('shows transaction ID in shortened format', () => {
+    renderComponent();
+    const txId = mockTransaction.id;
+    const shortenedTxId = screen.getByText(
+      `${txId.substring(0, 7)}...${txId.substring(txId.length - 5)}`,
+    );
+    expect(shortenedTxId).toBeInTheDocument();
+  });
+
+  it('displays network fee when present', () => {
+    renderComponent();
+    expect(screen.getByText('networkFee')).toBeInTheDocument();
+    expect(screen.getByText('1.0001 BTC')).toBeInTheDocument();
+  });
+
+  it('calls onClose when close button is clicked', () => {
+    renderComponent();
+    const closeButton = screen.getByRole('button', { name: /close/iu });
+    fireEvent.click(closeButton);
+    expect(mockProps.onClose).toHaveBeenCalled();
+  });
+
+  it('renders the view details button with correct link', () => {
+    renderComponent();
+    const viewDetailsButton = screen.getByText('viewDetails');
+    expect(viewDetailsButton).toBeInTheDocument();
+    fireEvent.click(viewDetailsButton);
+    expect(mockTrackEvent).toHaveBeenCalled();
+  });
+
+  it('handles different transaction statuses', () => {
+    const statuses = ['confirmed', 'pending', 'failed'];
+    statuses.forEach((status) => {
+      const propsWithStatus = {
+        ...mockProps,
+        transaction: {
+          ...mockTransaction,
+          status,
+        },
+      };
+      const { rerender } = renderComponent(propsWithStatus);
+      expect(
+        screen.getByText(status.charAt(0).toUpperCase() + status.slice(1)),
+      ).toBeInTheDocument();
+      rerender(<div />);
+    });
+  });
```

Comment:
IMO, it's better to use `it.each` here, in case 1 of the status is failing:

```suggestion
  // @ts-expect-error This is missing from the Mocha type definitions
  it.each(['confirmed', 'pending', 'failed'])
  ('handles different transaction status: %s', (status: string) => {
    const propsWithStatus = {
      ...mockProps,
      transaction: {
        ...mockTransaction,
        status,
      },
    };
    const { rerender } = renderComponent(propsWithStatus);
    expect(
      screen.getByText(status.charAt(0).toUpperCase() + status.slice(1)),
    ).toBeInTheDocument();
    rerender(<div />);
  });
```